packages/financial-templates-lib/src/logger/SlackTransport.ts

```diff
@@ -186,19 +192,32 @@ class SlackHook extends Transport {
       // If the log contains a notification path then use a custom slack webhook service. This lets the transport route to
       // different slack channels depending on the context of the log.
       const webhookUrl = this.escalationPathWebhookUrls[info.notificationPath] ?? this.defaultWebHookUrl;
-
-      const payload: { blocks?: Block[]; text?: string; mrkdwn?: boolean } = { mrkdwn: this.mrkdwn };
       const layout = this.formatter(info);
-      payload.blocks = layout.blocks || undefined;
-      // If the overall payload is less than 3000 chars then we can send it all in one go to the slack API.
-      if (JSON.stringify(payload).length < SLACK_MAX_CHAR_LIMIT) {
-        await this.axiosInstance.post(webhookUrl, payload);
-      } else {
-        // Iterate over each message to send and generate a axios call for each message.
-        for (const processedBlock of processMessageBlocks(payload.blocks)) {
-          payload.blocks = processedBlock;
-          await this.axiosInstance.post(webhookUrl, payload);
-        }
+
+      // Deliver in chunks if the the combined payload is more than SLACK_MAX_CHAR_LIMIT chars.
+      const payloads: (SlackFormatterResponse & { text?: string; mrkdwn?: boolean })[] =
+        JSON.stringify(layout.blocks).length < SLACK_MAX_CHAR_LIMIT
+          ? [{ blocks: layout.blocks, mrkdwn: this.mrkdwn }]
+          : processMessageBlocks(layout.blocks).map((blocks) => {
```

Comment:
Do we know the source of the 400 errors? Could they be because this function (erroneously) produces some blocks that have more characters than slack allows?

Reply:
As far as I can tell, the 400 error arises from the transport submitting badly formatted JSON.

I believe this is because the truncation logic that exists further down in this file is a bit too optimistic about being able to insert a "...truncated..." text string into the middle of a big batch of stringified JSON. It's unclear to me why we're apparently able to reconstruct that into a JSON object, so I think we do have follow-up work to be done there. But as a starting point, given that Slack specifically identify bad formatting as a source of 400 errors, it makes sense to me to start here and work backwards.

![2023-10-12-100354_1129x735_scrot](https://github.com/UMAprotocol/protocol/assets/108695806/05071bb9-13bf-4068-b41f-17fef8e5148b)